src/pages-and-resources/discussions/app-config-form/apps/legacy/LegacyConfigForm.test.jsx

```diff
@@ -192,4 +201,140 @@ describe('LegacyConfigForm', () => {
+    expect(queryByText(topic, 'It looks like this name is already in use')).toBeInTheDocument();
+  });
+
+  test('check duplicate error is removed on fields when name is fixed', async () => {
+    await mockStore(legacyApiResponse);
+    createComponent(defaultAppConfig);
+
+    const topicInput = createNewTopic('edx');
+    const topic = topicInput.closest('.collapsible-card');
+    topicInput.blur();
+    await waitForElementToBeRemoved(queryByText(topic, 'Choose a unique name for your topic'));
+    expect(queryByText(topic, 'It looks like this name is already in use')).toBeInTheDocument();
+
+    topicInput.setSelectionRange(0, topicInput.value.length);
+    userEvent.type(topicInput, 'valid');
+    expect(queryByText(topicInput, 'It looks like this name is already in use')).not.toBeInTheDocument();
+  });
+
+  test('check duplicate error is removed on deleting duplicate topic', async () => {
+    await mockStore(legacyApiResponse);
+    createComponent(defaultAppConfig);
+    const topicInput = createNewTopic('duplicate');
+    const topic = topicInput.closest('.collapsible-card');
+    const duplicateTopicInput = createNewTopic('duplicate');
+    const duplicateTopic = duplicateTopicInput.closest('.collapsible-card');
+    topicInput.blur();
+    duplicateTopicInput.blur();
+    await waitForElementToBeRemoved(queryByText(topic, 'Choose a unique name for your topic'));
+    expect(queryByText(topic, 'It looks like this name is already in use')).toBeInTheDocument();
+    expect(queryByText(duplicateTopic, 'It looks like this name is already in use')).toBeInTheDocument();
+    const deleteTopicButton = getByLabelText(duplicateTopic, 'Delete Topic');
+    userEvent.click(deleteTopicButton);
+    expect(duplicateTopic).not.toBeInTheDocument();
+    const delConfirmButton = getByText(container, 'Delete');
+    userEvent.click(delConfirmButton);
+    await waitForElementToBeRemoved(getByText(topic, 'It looks like this name is already in use'));
+    expect(queryByText(topic, 'It looks like this name is already in use')).not.toBeInTheDocument();
+  });
+
+  const cases = [['', 2], ['edx', 2], ['valid', 3]];
+  test.each(cases)(
+    'Check only valid topics are rendered',
+    async (topicName, expectedResult) => {
+      await mockStore(legacyApiResponse);
+      createComponent(defaultAppConfig);
+      userEvent.click(queryByText(container, 'Divide discussions by cohorts'));
+      userEvent.click(queryByText(container, 'Divide course-wide discussion topics'));
+      const topicList = getByRole(container, 'group');
+      const topicInput = createNewTopic(topicName);
+      const topic = topicInput.closest('.collapsible-card');
+      topicInput.focus();
+      topicInput.blur();
+      await waitForElementToBeRemoved(queryByText(topic, 'Choose a unique name for your topic'));
+      expect(topicList.getElementsByClassName('pgn__form-checkbox')).toHaveLength(expectedResult);
```

Comment:
the test shouldn't be based on the this class name, tomorrow if paragon changes this class, this test will start failing